armory/cli/tools/collect_outputs.py

```diff
@@ -0,0 +1,582 @@
+from armory.cli.tools.utils import _debug, simple_progress_bar
+from armory.logs import log, update_filters
+
+
+def _clean(output_dir: str, expiration: int = 6 * 60 * 60):
+    output_dir = Path(output_dir)
+    clean_dir = output_dir / ".cleaned"
+    # get all directories containing _only_ armory-log.txt and colored-log.txt
+    # ignore directories modified within the last 6 hours (to prevent active runs from being deleted)
+    empty_dirs = [
+        d
+        for d in output_dir.rglob("**/")
+        if d.name != "saved_samples"
+        and not d.is_relative_to(clean_dir)
+        and len(list(d.glob("*"))) == 2
+        and any([f.name == "armory-log.txt" for f in d.glob("*")])
+        and any([f.name == "colored-log.txt" for f in d.glob("*")])
+        and d.stat().st_mtime - time.time() > expiration
+    ]
+    log.info(
+        f"Found {len(empty_dirs)} empty directories to clean. Moving to {clean_dir}"
+    )
+    for d in empty_dirs:
+        target = clean_dir / d.relative_to(output_dir)
+        target.parent.mkdir(parents=True, exist_ok=True)
+        d.rename(target)
+    # delete all pairs of .png files NAME.png and NAME_adv.png where both files are equivalent
+    # (i.e. NAME.png is a copy of NAME_adv.png)
+    log.info("Cleaning up duplicate poisoned .png files")
+
+    pngs = list(
+        filter(
+            lambda p: len(list(p.parent.parent.glob("*Poison*"))) > 0,
+            output_dir.rglob("*x.png"),
+        )
+    )
+    log.debug(f"Found {len(pngs)} .png files to check for duplicates.")
+    unlinks = set()
+    for png in simple_progress_bar(pngs, msg="Checking equality..."):
+        adv = png.parent / (png.stem + "_adv.png")
+        if adv.exists() and Image.open(png).tobytes() == Image.open(adv).tobytes():
+            unlinks.add(png)
+            unlinks.add(adv)
+    log.info(
+        f"Found {len(pngs) - len(unlinks) // 2}/{len(pngs)} non-duplicate .png pairs. Moving the rest to {clean_dir}/pngs/"
+    )
+    for png in simple_progress_bar(unlinks, msg="Moving files..."):
+        target = clean_dir / png.relative_to(output_dir)
+        target.parent.mkdir(parents=True, exist_ok=True)
+        png.rename(target)
+    log.info(
+        f"Moved {len(unlinks) // 2} pairs of duplicate .png files to {clean_dir}/pngs/"
+    )
+
+
+def _get_run_name(d) -> str:
+    fallthrough = [
+        "config.sysconfig.filepath",
+        "config.sysconfig.config_filepath",
+        "config.sysconfig.output_filename",
+    ]
+    filepath = None
+    while filepath is None and len(fallthrough) > 0:
+        filepath = _get_dict_value(fallthrough.pop(0), d, default_value=None)
+    if filepath is None:
+        # can't find config filepath, just return result filepath as %s
+        return "%s"
+    p_filepath = Path(filepath)
+    if p_filepath.is_absolute():
+        common = os.path.commonpath([Path.cwd(), p_filepath])
+        filepath = p_filepath.relative_to(common)
+    return f"[{filepath}]({filepath})"
```

Comment:
Filepath can be `-` if config came from stdin; fixup what gets return in this case

Reply:
Addressed by https://github.com/twosixlabs/armory/pull/1986/commits/723c810c601b72f36d4db2bc5894b8f53692dcc0